tests/sentry/utils/sdk_crashes/test_event_stripper.py

```diff
@@ -0,0 +1,138 @@
+        }
+
+        for key in keys_kept:
+            assert stripped_event_data.get(key) is not None, f"key {key} should be kept"
+
+    def test_strip_event_data_without_debug_meta(self):
+        event_data = get_crash_event()
+        event_data["debug_meta"]["images"] = None
+
+        event = self.create_event(
+            data=event_data,
+            project_id=self.project.id,
+        )
+
+        stripped_event_data = strip_event_data(event, CocoaSDKCrashDetector())
+
+        debug_meta_images = get_path(stripped_event_data, "debug_meta", "images")
+        assert debug_meta_images is None
+
+    def test_strip_event_data_strips_context(self):
+        event = self.create_event(
+            data=get_crash_event(),
+            project_id=self.project.id,
+        )
+
+        stripped_event_data = strip_event_data(event, CocoaSDKCrashDetector())
+
+        contexts = stripped_event_data.get("contexts")
+        assert contexts is not None
+        assert contexts.get("app") is None
+        assert contexts.get("os") is not None
+        assert contexts.get("device") is not None
+
+    def test_strip_event_data_strips_non_referenced_dsyms(self):
+        event = self.create_event(
+            data=get_crash_event(),
+            project_id=self.project.id,
+        )
+
+        stripped_event_data = strip_event_data(event, Mock())
+
+        debug_meta_images = get_path(stripped_event_data, "debug_meta", "images")
+
+        image_addresses = set(map(lambda image: image["image_addr"], debug_meta_images))
+        expected_image_addresses = {"0x1a4e8f000", "0x100304000", "0x100260000"}
+        assert image_addresses == expected_image_addresses
+
+    def test_strip_frames_sentry_in_app_frame_kept(self):
+        frames = get_frames("SentryCrashMonitor_CPPException.cpp", sentry_frame_in_app=True)
+        self._execute_strip_frames_test(frames)
+
+    def test_strip_frames_sentry_non_in_app_frame_kept(self):
+        frames = get_frames("SentryCrashMonitor_CPPException.cpp", sentry_frame_in_app=False)
+        self._execute_strip_frames_test(frames)
+
+    def _execute_strip_frames_test(self, frames):
+        event_data = get_crash_event_with_frames(frames)
+
+        event = self.create_event(
+            data=event_data,
+            project_id=self.project.id,
+        )
+
+        stripped_event_data = strip_event_data(event, CocoaSDKCrashDetector())
+
+        stripped_frames = get_path(
+            stripped_event_data, "exception", "values", -1, "stacktrace", "frames"
+        )
+
+        assert len(stripped_frames) == 6
+        assert (
+            len(
+                [
+                    frame
+                    for frame in stripped_frames
+                    if frame["function"] == IN_APP_FRAME["function"]
+                ]
+            )
+            == 0
+        ), "in_app frame should be removed"
+
+
+@region_silo_test
+class EventStripperTest(
+    TestCase,
+    EventStripperTestMixin,
+):
```

Comment:
this "write a bunch of mixins" pattern is very strange -- I would not recommend it (where did you pick up this pattern?)

the recommended form of tests for pytest is to write function tests -- you can have many functions.   you can also write many classes (you don't need just one) but it is not recommended to write class-based tests

Reply:
Initially, I had function-based tests. When I put it into [post_process.py](https://github.com/getsentry/sentry/blob/b953b0911253e786d8e4541b6f8a08feeab167fb/src/sentry/tasks/post_process.py) it didn't work correctly. So I changed the code to accept events as they are passed around in post_process.py. Then I looked at the [test_post_process.py](https://github.com/getsentry/sentry/blob/b953b0911253e786d8e4541b6f8a08feeab167fb/tests/sentry/tasks/test_post_process.py) and copied the pattern to make it work. If you have any suggestions to improve this, please let me know. I'm also not a fan of this pattern.